app/middlewares/session.js

```diff
@@ -0,0 +1,31 @@
+const bcrypt = require('bcrypt');
+const util = require('util');
+
+const { findUserByEmail } = require('../services/user');
+const { sessionError } = require('../errors');
+
+const findUser = (request, _response, next) => {
+  // eslint-disable-next-line consistent-return
+  findUserByEmail(request.body.email).then(user => {
```

Comment:
you are never calling next if the db call fails. This can be easily solved with a `.catch` at the end or a try/catch
```
const findUser = async (request, _response, next) => {
  try {
    const user = await findUserByEmail(request.body.email);
    if (!user) throw etc
    next();
  } catch (err) {
    next(err)
  }
};
```